example/src/main/tut/typeclass/index.md

```diff
@@ -4,3 +4,81 @@ title:  "Type classes"
 ---
 
 # Type classes
+
+_In computer science, a type class is a type system construct that supports ad hoc polymorphism.
+This is achieved by adding constraints to type variables in parametrically polymorphic types._ <sup>[1](#f1)</sup>
+
+Type class instances allow us to add behaviour to types without changing the types themselves.
+
+As an example, let us look at the [Debug](./Debug.html) type class, a safe alternative to the JVM's `Object.toString`.
+It defines that all its instances implement a method `debug` with the following signature:
+
+```tut:silent
+trait Debug[A] {
+  def debug(a: A): String
+}
+```
+
+Now assume that we have a class which we are unable or unwilling to change in order to give it a conversion into a `String` representation.
+
+```tut:silent
+final case class BusinessObject(id: Long, value: Long)
+```
+
+Using the `Debug` type class, we can easily just define such a conversion:
+
+```tut:silent
+implicit val businessObjectDebug: Debug[BusinessObject] = new Debug[BusinessObject] {
+  def debug(b: BusinessObject) = s"BO[${b.id} = ${b.value}]"
+}
+```
+
+Now, whenever we would like to turn our `BusinessObject` into a String, we can do so using our `Debug` instance.
+
+```tut
+val bo = BusinessObject(1234L, 1234567L)
+
+businessObjectDebug.debug(bo)
+```
+
+However, this is not too convenient and very specific to our `BusinessObject`. Would it not be nicer to be able to say `bo.debug`?
+
+Luckily, **Scalaz** makes all this very easy.
+
+We need something similar to the following:
```

Comment:
Not entirely sure if this is the right place to explain syntax enrichment :) Have you considered something along the lines of:

> However, this is not too convenient and very specific to our `BusinessObject`. Would it not be nicer to be able to say `bo.debug`? By default, Scalaz enriches all objects `o` that have an instance of typeclass `Debug` with a method `o.debug`:

and then continue with the `BusinessObject(1234L, 1234567L).debug` snippet.

Reply:
AAh, I see. This is not `Debug` specific, my mistake :smile_cat: